pom.xml

```diff
@@ -210,26 +237,66 @@
 
             <!-- Git describe -->
             <plugin>
-                <groupId>com.lukegb.mojo</groupId>
-                <artifactId>gitdescribe-maven-plugin</artifactId>
-                <version>3.0</version>
+                <groupId>pl.project13.maven</groupId>
+                <artifactId>git-commit-id-plugin</artifactId>
+                <version>2.2.4</version>
                 <executions>
                     <execution>
-                        <id>git-describe</id>
-                        <phase>initialize</phase>
+                        <id>get-the-git-infos</id>
                         <goals>
-                            <goal>gitdescribe</goal>
+                            <goal>revision</goal>
                         </goals>
-                        <configuration>
-                            <descriptionProperty>buildHead</descriptionProperty>
-                            <extraArguments>
-                                <param>--tags</param>
-                                <param>--always</param>
-                                <param>HEAD</param>
-                            </extraArguments>
-                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>validate-the-git-infos</id>
+                        <goals>
+                            <goal>validateRevision</goal>
+                        </goals>
+                        <phase>package</phase>
                     </execution>
                 </executions>
+                <configuration>
+                    <dotGitDirectory>${project.basedir}/.git</dotGitDirectory>
+                    <prefix>git</prefix>
+                    <dateFormat>yyyy-MM-dd HH:mm:ss</dateFormat>
+                    <verbose>false</verbose>
+                    <generateGitPropertiesFile>true</generateGitPropertiesFile>
+                    <generateGitPropertiesFilename>${basedir}/src/main/resources/git.properties</generateGitPropertiesFilename>
+                    <format>properties</format>
+                    <failOnNoGitDirectory>true</failOnNoGitDirectory>
+                    <failOnUnableToExtractRepoInfo>false</failOnUnableToExtractRepoInfo>
+                    <excludeProperties>
+                        <excludeProperty>git.branch</excludeProperty>
```

Comment:
Considering we also have an `includeOnlyProperties`, is this list not redundant. It seems to me like there must be a better way to only include the right properties.

Reply:
I've just done some testing and it looks like that is the case, it adds a bit more info to the git.properties file but nothing too major.